daemon/container.go

```diff
@@ -1627,3 +1618,53 @@ func (container *Container) monitorExec(execConfig *execConfig, callback execdri
 
 	return err
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	paths, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(paths) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := findSignificantParent(paths)
+
+	relative := make([]string, len(paths))
+
+	p := fmt.Sprintf("%s%s", baseDir, string(filepath.Separator))
```

Comment:
I think you need an if-statement around the adding of "/" so that if baseDir is just "/" you don't add another one.